frontend/src/entities/batch-translate/create/batch-translate-create.tsx

```diff
@@ -263,19 +193,24 @@ export function BatchTranslateCreate () {
                 })
             );
 
-            if (response.type.endsWith('/fulfilled')) {
+            if (isFulfilled(response)) {
                 notificationService.generateNotification(
                     `Successfully created translation job ${state.form.JobName}.`,
                     'success'
                 );
-                const newDataset = {
-                    name: state.form.OutputDataset.Name,
-                    description: `Dataset created as part of the Batch Translate job: ${state.form.JobName}`,
-                    type: state.form.OutputDataset.Type,
-                    scope: determineScope(state.form.OutputDataset.Type, projectName, userName!)
-                } as IDataset;
+                
+                const dataset = datasetFromS3Uri(state.form.OutputDataConfig.S3Uri);
+                if (dataset) {
+                    const newDataset = {
+                        name: dataset.name,
+                        description: `Dataset created as part of the Batch Translate job: ${state.form.JobName}`,
+                        type: dataset.type,
+                        scope: dataset.scope
+                    } as IDataset;
+    
+                    createDatasetHandleAlreadyExists(newDataset);
```

Comment:
This may be from Doug's previous changes, but I feel like a better name for this function is something like `useOrCreateDataset`

Reply:
It was the existing name but I agree that the suggestion name would be clearer. I'll roll in that change with all the other change requests from this PR.